ui/src/app/utils/index.tsx

```diff
@@ -387,3 +388,29 @@ export function resettableTimeout(f, timeoutInSeconds) {
     getTimer: () => timeout
   };
 }
+
+export function highlightSearchTerm(searchTerm: string, stringToHighlight: string) {
+  if (searchTerm === '' || searchTerm.length < 3) {
+    return stringToHighlight;
+  }
+  const words: string[] = [];
+  const matchString = new RegExp(searchTerm.trim(), 'i');
+  const matches = stringToHighlight.match(new RegExp(matchString, 'gi'));
+  const splits = stringToHighlight.split(new RegExp(matchString, 'gi'));
+  if (matches) {
+    for (let i = 0; i < matches.length; i++) {
+      words.push(splits[i], matches[i]);
+    }
+    words.push(splits[splits.length - 1]);
+  } else {
+    words.push(splits[0]);
+  }
+  return words.map((word, w) => <span key={w}
+    style={matchString.test(word.toLowerCase()) ? {
+      color: colorWithWhiteness(colors.success, -0.4),
```

Comment:
I would make the highlight color parameterizeable as well, so that users can pick what colors they want.

Reply:
So we'd pass in a single color, then still use `colorWithWhiteness` to get values for `color` and `backgroundColor`? Maybe something like
```
export function highlightSearchTerm(searchTerm: string, stringToHighlight: string, highlightColor: string) {
...
return words.map((word, w) => <span key={w}
    style={matchString.test(word.toLowerCase()) ? {
      color: colorWithWhiteness(highlightColor, -0.4),
      backgroundColor: colorWithWhiteness(highlightColor, 0.7),
      display: 'inline-block'
    } : {}}>
      {word}
    </span>);
}
```

Reply:
That seems like a reasonable way to do it. I could see passing in both colors, but this seems fine for flexibility.